src/controllers/articleController.js

```diff
@@ -81,15 +81,75 @@ class ArticleController {
         ]
       });
       if (!articles.length) {
-        return res.status(200).json({
+        return res.status(404).json({
           message: 'No articles found at the moment! please come back later'
         });
       }
       res.status(200).json({
         articles
       });
     } catch (err) {
-      throw (err);
+      throw err;
+    }
+  }
+
+  /**
+   * @description update article
+   * @param {object} req
+   * @param {object} res
+   * @return {object} return object with updated article.
+   */
+  static async updateArticle(req, res) {
+    try {
+      const article = {};
+      const {
+        title, description, body, tags, category
+      } = req.body;
+      const originalArticle = await Article.findOne({ where: { slug: req.userData.slug } });
+      if (title) {
+        article.title = title.trim();
+        article.slug = slugGen(title.trim());
+      }
+      article.description = (description || originalArticle.description).trim();
+      article.body = (body || originalArticle.body).trim();
+      article.tagList = (tags || originalArticle.tagList.toString()).trim().split(/[ ,]+/);
+      article.category = (category || originalArticle.category).trim();
```

Comment:
why trim `originalArticle.category`?

Reply:
Here I am checking for the category from req.body to trim but when I define (category.trim() || originalArticle.category)  throws an error when the category is undefined